enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -398,14 +404,12 @@ def catalog_contains_content_key(self, content_key):
             content_key,
         )
 
+    @memoize_instance_method
```

Comment:
Subtle change, but take note - the result of this method is now memoized for the lifetime of the policy instance. This is what necessitated some of the additional `tearDown()` logic in the model tests below.

Reply:
Can you explain the reasoning for memoizing this method?  If the purpose is to avoid redundant network calls to memcache during the lifecycle of this request, then my understanding is that TieredCache already handles that under the hood:

> With a tiered cache, multiple calls for the same value within a request will not require additional network calls to Memcached.

https://docs.openedx.org/projects/openedx-proposals/en/latest/best-practices/oep-0022-bp-django-caches.html

Reply:
Ah, good point, that's not necessary, wasn't thinking about the fact that this was already Tiered-cached.

Reply:
And on second though, it seems dangerous to introduce that memoization without some notion of a cache key (for indexing on the input of the function). I'll rip that out and use a RequestCache.

Reply:
when I brought up TieredCache was referring to what's already being used by get_and_cache_content_metadata() https://github.com/openedx/enterprise-access/blob/cc341c005bc9b3ca9cdff9fa9a252b1bc7800204/enterprise_access/apps/subsidy_access_policy/content_metadata_api.py#L31